packages/expect/src/types.ts

```diff
@@ -324,6 +324,7 @@ export interface Matchers<R extends void | Promise<void>, T = unknown> {
    */
   toMatchObject(
     expected: Record<string, unknown> | Array<Record<string, unknown>>,
+    options?: jestMatcherUtils.MatchObjectOptions,
```

Comment:
Fine to repeat it. There are type tests to catch any issues.

```suggestion
    options?: {ignoreElementsOrder?: boolean},
```